src/components/Subscription.ts

```diff
@@ -0,0 +1,22 @@
+import { ReactNode } from 'react';
+
+import {
+  useSubscription,
+  UseSubscriptionArgs,
+  UseSubscriptionState,
+  SubscriptionHandler,
+} from '../hooks';
+
+export interface SubscriptionProps<T, V, R = T> extends UseSubscriptionArgs<V> {
+  handler?: SubscriptionHandler<T, R>;
+  children: (arg: UseSubscriptionState<R>) => ReactNode;
+}
+
+export function Subscription<T = any, R = T, V = any>({
+  children,
+  handler,
+  ...args
+}: SubscriptionProps<T, V, R>): ReactNode {
```

Comment:
Super nit – it looks like `SubscriptionProps` receives its generics in the order `<T, V, R>`, but `Subscription`'s generics (and `useSubscription`) are in the order `<T, R, V>`. I'd keep the ordering consistent if possible here just to keep these signatures easy to read.

Reply:
Good catch! 👍